common/challenge.mk

```diff
@@ -0,0 +1,23 @@
+INTERNAL_PORT := 80
+CONT_NAME := ${IMG_NAME}
+FILE_FLAG := ../flag
+FLAG := $(shell cat $(FILE_FLAG))
+
+run: generate build
+	docker run -d -p $(EXTERNAL_PORT):$(INTERNAL_PORT) --name $(CONT_NAME) -t $(IMG_NAME)
+
+build: generate
+	docker build -t $(IMG_NAME) -f Dockerfile ..
+
+generate:
```

Comment:
In my opinion, this should call the `Makefile` in `src/`, if available. Something like:
```
test -d src/Makefile && make -C src/
```

Reply:
Done.